x-element.js

```diff
@@ -1,112 +1,176 @@
 /** Base element class for creating custom elements. */
 export default class XElement extends HTMLElement {
-  /** Extends HTMLElement.observedAttributes to handle the properties block. */
+  /**
+   * Extends HTMLElement.observedAttributes to handle the properties block.
+   * @returns {string[]}
+   */
   static get observedAttributes() {
     XElement.#analyzeConstructor(this);
     return [...XElement.#constructors.get(this).attributeMap.keys()];
   }
 
-  /** Default templating engine. Use "templateEngine" to override. */
+  /**
+   * Default templating engine. Use "templateEngine" to override.
+   * @returns {{[key: string]: Function}}
+   */
   static get defaultTemplateEngine() {
     return TemplateEngine.interface;
   }
 
-  /** Configured templating engine. Defaults to "defaultTemplateEngine".
-   *
+  /**
+   * Configured templating engine. Defaults to "defaultTemplateEngine".
    * Override this as needed if x-element's default template engine does not
    * meet your needs. A "render" method is the only required field. An "html"
    * tagged template literal is expected, but not strictly required.
+   * @returns {{[key: string]: Function}}
    */
   static get templateEngine() {
     return XElement.defaultTemplateEngine;
   }
 
+  /**
+   * Observe callback.
+   * @callback observeCallback
+   * @param {HTMLElement} host
+   * @param {any} value
+   * @param {any} oldValue
+   */
+
+  /**
+   * A property value.
+   * @typedef {object} Property
+   * @property {any} [type]
+   * @property {string} [attribute]
+   * @property {string[]} [input]
+   * @property {Function} [compute]
+   * @property {observeCallback} [observe]
+   * @property {boolean} [reflect]
+   * @property {boolean} [internal]
+   * @property {boolean} [readOnly]
+   * @property {any|Function} [initial]
+   * @property {any|Function} [default]
+   */
+
   /**
    * Declare watched properties (and related attributes) on an element.
-   *
-   *   static get properties() {
-   *     return {
-   *       property1: {
-   *         type: String,
-   *       },
-   *       property2: {
-   *         type: Number,
-   *         input: ['property1'],
-   *         compute: this.computeProperty2,
-   *         reflect: true,
-   *         observe: this.observeProperty2,
-   *         default: 0,
-   *       }
-   *     };
-   *   }
+   * ```js
```

Comment:
JSDoc also has an `@example` tag… but this feels more natural to write code blocks like this and it appears to be supported by JSR.